setup.py

```diff
@@ -30,7 +30,7 @@
     url='https://github.com/bigscience-workshop/promptsource.git',
     author='BigScience - Prompt Engineering Working Group',
     author_email='sbach@cs.brown.edu,victor@huggingface.co',
-    python_requires='>=3.7, <3.8',
+    python_requires='>=3.7',
```

Comment:
A python-version-related suggestion:

```python
requirements = [
    "streamlit==0.82",
    "datasets>=1.7.0",  # For this PR alone we need "<2" and yet we can just rebased onto `main` for #735.
    "pandas<1.4; python_version<'3.8'",  # Set pandas version for Python 3.7.
    "pandas>=1.4; python_version>='3.8'",
    "plotly",
    ##############################################################
    # Dependencies in this section are added for specific datasets
    ##############################################################
    "py7zr",
    ##############################################################
    # End of dataset-specific dependencies
    ##############################################################
    # TODO: We may consider moving this to a `dev` of `extras` and then changing GH Actions with `pip install .[dev]`.
    "black<=21.12b0",
    "flake8",
    "isort==5.8.0",
    "pytest",
]
```

Because `streamlit`, `datasets`, `plotly`, and `py7zr` will cover everything else, except that `pandas` needs a switch for Python 3.7.